src/iptc.cpp

```diff
@@ -345,7 +346,10 @@ namespace Exiv2 {
 
     void IptcData::printStructure(std::ostream& out, const Slice<byte*>& bytes, uint32_t depth)
     {
-        uint32_t i = 0;
+        if (bytes.size() < 3) {
+            return;
+        }
+        size_t i = 0;
         while (i < bytes.size() - 3 && bytes.at(i) != 0x1c)
```

Comment:
The type of [`bytes.size()`](https://github.com/Exiv2/exiv2/blob/fec1813a3e7b0e4aaed7fae7ba9f9b49ca526638/include/exiv2/slice.hpp#L98) is `size_t`, so `size_t` is a better choice for `i`.